### PR TITLE
add typegurad and types

### PR DESCRIPTION
**Pull request by autoih**
Hopefully this PR helps, linked to #894.

### CODE REVIEW

**Comment by seanpmorgan** [comment]
Thanks @autoih!

Would you mind removing these from the exception list:
https://github.com/tensorflow/addons/blob/master/tools/ci_build/verify/check_typing_info.py#L27

**Comment by autoih** [comment]
Thanks @seanpmorgan. I also removed some (maxout, gelu, etc) have been updated.

File: tensorflow_addons/layers/poincare.py [hunk 1]
```diff
@@ -41,7 +42,8 @@ class PoincareNormalize(tf.keras.layers.Layer):
         numerical stability.
     """
 
-    def __init__(self, axis=1, epsilon=1e-5, **kwargs):
+    @typechecked
+    def __init__(self, axis: int = 1, epsilon: int = 1e-5, **kwargs):
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/layers/poincare.py]:
```suggestion
    def __init__(self, axis: int = 1, epsilon: float = 1e-5, **kwargs):
```

Comment (autoih) [reply tensorflow_addons/layers/poincare.py]: Thank you, @gabrieldemarmiesse.

**Comment by gabrieldemarmiesse** [comment]
Nice :)

**Pull request merged**